### PR TITLE
Fix a few issues with certs

### PR DESCRIPTION
**Pull request by nathanleclaire**
- regenerate-certs was busted, because of recent changes I made to
  Provision() method
- checksum was different between local certs and remote certs

This PR fixes both issues and adds tests to verify that both conditions
are checked in the future.

Closes https://github.com/docker/machine/issues/1144.
Closes https://github.com/docker/machine/issues/1159.

cc @ehazlett @hairyhenderson for review.

Assuming this gets merged, I will update the integration tests refactor to move these tests over to the new format.

Signed-off-by: Nathan LeClaire <nathan.leclaire@gmail.com>

### CODE REVIEW

File: test/integration/certs-checksum.bats [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+export DRIVER=virtualbox
+export NAME="bats-$DRIVER-test"
+export MACHINE_STORAGE_PATH=/tmp/machine-bats-test-$DRIVER
+
+@test "$DRIVER: create" {
+  run machine create -d $DRIVER $NAME
+}
+
+@test "$DRIVER: verify that server cert checksum matches local checksum" {
```
Comment (nathanleclaire) [inline test/integration/certs-checksum.bats]: This is the most beautiful integration test ever in the world

File: test/integration/certs-checksum.bats [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  # TODO: Does this test work OK on Linux? cc @ehazlett
+  # Have to create this directory and file or else the OpenSSL checksum will barf.
+  machine ssh $NAME -- sudo mkdir -p /usr/local/ssl
+  machine ssh $NAME -- sudo touch /usr/local/ssl/openssl.cnf
+  SERVER_CHECKSUM=$(machine ssh $NAME -- openssl dgst -sha256 /var/lib/boot2docker/ca.pem | awk '{ print $2 }')
```
Comment (nathanleclaire) [inline test/integration/certs-checksum.bats]: wow

File: test/integration/certs-checksum.bats [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  LOCAL_CHECKSUM=$(openssl dgst -sha256 $MACHINE_STORAGE_PATH/certs/ca.pem | awk '{ print $2 }')
```
Comment (nathanleclaire) [inline test/integration/certs-checksum.bats]: such `awk`

File: test/integration/certs-checksum.bats [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  echo ${SERVER_CHECKSUM}
+  echo ${LOCAL_CHECKSUM}
+  [[ ${SERVER_CHECKSUM} == ${LOCAL_CHECKSUM} ]]
```
Comment (nathanleclaire) [inline test/integration/certs-checksum.bats]: very checksum

File: test/integration/regenerate-certs.bats [hunk 5]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bats
```
Comment (nathanleclaire) [inline test/integration/regenerate-certs.bats]: This test just makes sure we don't inadvertently bust `regenerate-certs` again.

File: test/integration/certs-checksum.bats (hunk 4 above)
Comment (hairyhenderson) [inline test/integration/certs-checksum.bats]: lol

**Comment by hairyhenderson** [comment]
:+1: LGTM!

**Comment by ehazlett** [comment]
LGTM

**Comment by ehazlett** [comment]
img

**Pull request merged**